### DISCUSSION TITLE
Split self-hosted runners permission from "Repository Administration" for personal account

### DISCUSSION THREAD

**Discussion by sksat** [community/community#36583] [Actions]
In organization account, we can control self-hosted runner permission individually like below.
![2022-10-19_03-50](https://user-images.githubusercontent.com/23310673/196518303-36151dc9-da46-4612-906f-739bafd9f539.png)

Then, if we want to do this personal account, we have to select "Repository Administration" permission.
Registering self-hosted runner is certainly a "Repository Settings", but it should be permit individually.
"Administration" permission is too large permission for this use case.

**Comment by AF250329** [comment]
Hi

1) I don't understand: `Then, if we want to do this personal account, we have to select "Repository Administration" permission.`  <-- what do you mean ?

2) In certain cases creating self-hosted runner and receiving all the code and actually even building it on hostile machine could lead to great security hole in repo. So I guess this is the reason that only repo Administrators could do that.

Sorry - don't really understand your question... Could you explain it like I am 6 year old ?

**Reply by sksat** [reply]
It means we want to register/unregister self-hosted Actions runner in personal account's repository like in organizations.

Indeed, it is understandable that in many cases, repository administrators often register self-hosted runners.
However, I think this is only historical reason and does not seem to be a necessity (in fact, organisations have separate permissions).
As a specific use case, it is very incompatible with automation such as [actions-runner-controller](https://github.com/actions-runner-controller/actions-runner-controller).
Sure, I am the repository administrator for my account's repository, but that doesn't mean I want to give this controller permission to "Repository Administration" permission(It can delete repository!). This is clearly excessive permission.

**Reply by sksat** [reply]
Also, register self-hosted runner does not mean that you can read the contents of the repository.
It can and should be controlled by the 'Contents' permission.

**Comment by ossanna16** [comment]
Hey there @sksat and thank you for asking a good question! I'm glad you found our community 🙂

**Comment by duilio** [comment]
Hi, just came here to raise the same issue and found this one already submitted. I saw Github improved a lot the permission system in the last years making changes to make them more granular, I was wondering if there is something in the roadmap here as well.
